### ISSUE TITLE
TypeError: is of type RuntimeBatchRequest which cannot be serialized. when trying to run checkpoint with runtime batches

### ISSUE THREAD

**Issue by HimanshuPatri7** [fivetran/great_expectations#3497]
I am trying to run a checkpoint with batches passed as runtime batches and receiving an error saying 'RuntimeBatchRequest' object has no attribute 'get'.
I have tried to create checkpoint config in python using jupyter notebook.

1. Created a Runtimebatch for a pandas dataframe. I am trying to create multiple batches using a loop and storing in dict.  Sheets is a list containing sheet names of an excel. 

```
for sheet in sheets:
 batches[sheet] = RuntimeBatchRequest(
                        datasource_name="my_runtime_pandas_data",
                        data_connector_name="my_runtime_data_connector",
                        data_asset_name=f"{sheet}_Batch",
                        runtime_parameters={
                          "batch_data": df_sheets[sheet]
    },
    batch_identifiers={
        "sheet_name" : sheet
    }
)
```

2. Successfully created a validator head using a single batch from the dict of batches.
3. When creating a checkpoint using below code, I am facing an error saying:

**Code for creating checkpoint and opening data docs:**

```
checkpoint_config = {
    "class_name": "SimpleCheckpoint",
    "validations": [
        {
            "batch_request": batches[schema],
            "expectation_suite_name": expectation_suite_name
        }
    ]
}
checkpoint = SimpleCheckpoint(
    f"_tmp_checkpoint_{expectation_suite_name}",
    context,
    **checkpoint_config
)

checkpoint_result = checkpoint.run()
context.build_data_docs()
validation_result_identifier = checkpoint_result.list_validation_result_identifiers()[0]
context.open_data_docs(resource_identifier=validation_result_identifier)

```
**The Error:**
 
TypeError: {
  "datasource_name": "my_runtime_pandas_data",
  "data_connector_name": "my_runtime_data_connector",
  "data_asset_name": "NSS_Batch",
  "batch_identifiers": {
    "sheet_name": "NSS"
  },
  "runtime_parameters": {
    "batch_data": "<class 'pandas.core.frame.DataFrame'>"
  }
} is of type RuntimeBatchRequest which cannot be serialized.

Expected Behaviour:

I am trying to create checkpoints to validate multiple batches passed at runtime and storing the validation results for each and generating their data docs.

**Environment (please complete the following information):**
 - Operating System: [Windows]
 - Great Expectations Version: [ 0.13.34]

**Entire Error**
```
TypeError                                 Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_17888/3477869561.py in <module>
     11     f"_tmp_checkpoint_{expectation_suite_name}",
     12     context,
---> 13     **checkpoint_config
     14 )
     15 

C:\Program Files\Python37\lib\site-packages\great_expectations\checkpoint\checkpoint.py in __init__(self, name, data_context, config_version, template_name, module_name, class_name, run_name_template, expectation_suite_name, batch_request, action_list, evaluation_parameters, runtime_configuration, validations, profilers, validation_operator_name, batches, ge_cloud_id, site_names, slack_webhook, notify_on, notify_with, expectation_suite_ge_cloud_id, **kwargs)
    749             notify_with=notify_with,
    750             ge_cloud_id=ge_cloud_id,
--> 751             expectation_suite_ge_cloud_id=expectation_suite_ge_cloud_id,
    752         ).build()
    753 

C:\Program Files\Python37\lib\site-packages\great_expectations\checkpoint\configurator.py in build(self)
    113         self._validate_slack_configuration()
    114 
--> 115         return self._build_checkpoint_config()
    116 
    117     def _build_checkpoint_config(self) -> CheckpointConfig:

C:\Program Files\Python37\lib\site-packages\great_expectations\checkpoint\configurator.py in _build_checkpoint_config(self)
    144 
    145         logger.debug(
--> 146             f"SimpleCheckpointConfigurator built this CheckpointConfig:"
    147             f" {json.dumps(checkpoint_config.to_json_dict(), indent=4)}"
    148         )

C:\Program Files\Python37\lib\site-packages\great_expectations\data_context\types\base.py in to_json_dict(self)
    114         """
    115         commented_map: CommentedMap = self.commented_map
--> 116         return convert_to_json_serializable(data=commented_map)
    117 
    118     @property

C:\Program Files\Python37\lib\site-packages\great_expectations\core\util.py in convert_to_json_serializable(data)
    143         for key in data:
    144             # A pandas index can be numeric, and a dict key can be numeric, but a json key must be a string
--> 145             new_dict[str(key)] = convert_to_json_serializable(data[key])
    146 
    147         return new_dict

C:\Program Files\Python37\lib\site-packages\great_expectations\core\util.py in convert_to_json_serializable(data)
    150         new_list = []
    151         for val in data:
--> 152             new_list.append(convert_to_json_serializable(val))
    153 
    154         return new_list

C:\Program Files\Python37\lib\site-packages\great_expectations\core\util.py in convert_to_json_serializable(data)
    143         for key in data:
    144             # A pandas index can be numeric, and a dict key can be numeric, but a json key must be a string
--> 145             new_dict[str(key)] = convert_to_json_serializable(data[key])
    146 
    147         return new_dict

C:\Program Files\Python37\lib\site-packages\great_expectations\core\util.py in convert_to_json_serializable(data)
    236         raise TypeError(
    237             "%s is of type %s which cannot be serialized."
--> 238             % (str(data), type(data).__name__)
    239         )
    240 

TypeError: {
  "datasource_name": "my_runtime_pandas_data",
  "data_connector_name": "my_runtime_data_connector",
  "data_asset_name": "NSS_Batch",
  "batch_identifiers": {
    "sheet_name": "NSS"
  },
  "runtime_parameters": {
    "batch_data": "<class 'pandas.core.frame.DataFrame'>"
  }
} is of type RuntimeBatchRequest which cannot be serialized.
```

**Comment by NathanFarmer** [comment]
Hi @HimanshuPatri7! Thanks for opening this issue. I have opened PR #3502 to address this bug. I will follow-up here once the fix has been reviewed and merged.

**Comment by NathanFarmer** [comment]
Hi @HimanshuPatri7, this has been addressed by PR #3502 and the fix will be included in release 0.13.38.
